certs-manager/src/main/java/io/strimzi/certs/OpenSslCertManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * An OpenSSL based certificates manager
+ */
+public class OpenSslCertManager implements CertManager {
+
+    private static final Logger log = LogManager.getLogger(OpenSslCertManager.class);
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, int days) throws IOException {
+        generateSelfSignedCert(keyFile, certFile, null, days);
+    }
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, Subject sbj, int days) throws IOException {
+
+        List<String> cmd = new ArrayList<>(Arrays.asList("openssl", "req", "-x509", "-new", "-days", String.valueOf(days), "-batch", "-nodes",
+                "-out", certFile.getAbsolutePath(), "-keyout", keyFile.getAbsolutePath()));
+
+        if (sbj != null) {
+            cmd.addAll(Arrays.asList("-subj", sbj.toString()));
+        }
+
+        exec(cmd);
+    }
+
+    @Override
+    public void generateCsr(File keyFile, File csrFile, Subject sbj) throws IOException {
+        exec("openssl", "req", "-new", "-batch", "-nodes",
+                "-keyout", keyFile.getAbsolutePath(), "-subj", sbj.toString(), "-out", csrFile.getAbsolutePath());
+    }
+
+    @Override
+    public void generateCert(File csrFile, File caKey, File caCert, File crtFile, int days) throws IOException {
+        exec("openssl", "x509", "-req", "-days", String.valueOf(days), "-in", csrFile.getAbsolutePath(),
+                "-CA", caCert.getAbsolutePath(), "-CAkey", caKey.getAbsolutePath(), "-CAcreateserial",
+                "-out", crtFile.getAbsolutePath());
+    }
+
+    private void exec(String... cmd) throws IOException {
+        exec(Arrays.asList(cmd));
+    }
+
+    private void exec(List<String> cmd) throws IOException {
```

Comment:
This is now sync operation. Do we want to make it async outside of these classes? Or should we do it asynchronously already here? Did you had some thoughts about it?

Reply:
Yes. My initial thought was to make it asynchronous using Vert.x as we already have done in other components; then my concern was about adding the Vert.x dependency for this component as well maybe not needed. The other solution could be using "old" school async with threads. Finally, decide to leave as it is for now and see. Maybe starting to use Vert.x should not be a problem considering that we and EnMasse are using it in all the other places?

Reply:
I mean ... saying that we do it on the outside is valid point is valid response. I was pretty much wondering what your plans were. The argument about introducing Vert.x is probably valid.